beacon-chain/sync/checkpoint/common.go

```diff
@@ -0,0 +1,28 @@
+package checkpoint
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/db"
+)
+
+// Initializer describes a type that is able to obtain the checkpoint sync data (BeaconState and SignedBeaconBlock)
+// in some way and perform database setup to prepare the beacon node for syncing from the given checkpoint.
+// See FileInitializer and APIInitializer.
+type Initializer interface {
+	Initialize(ctx context.Context, d db.Database) error
+}
+
+// isCheckpointStatePresent checks if the checkpoint and corresponding state exist in the database.
+func isCheckpointStatePresent(ctx context.Context, d db.Database) (bool, error) {
+	origin, err := d.OriginCheckpointBlockRoot(ctx)
+	if err != nil && !errors.Is(err, db.ErrNotFound) {
+		return false, errors.Wrap(err, "error while checking database for origin root")
+	}
```

Comment:
What is the intended return value for `err != nil && errors.Is(err, db.ErrNotFound)` but `d.HasState(ctx, origin)` condition?

Reply:
It seems I need a fix here.
If `errors.Is(err, db.ErrNotFound)` is true, `origin` will be `nil`. `HasState` should not receive nil value, so in this case it should return `false, nil`.
Thanks a lot!